src/lib/redux/inventoryData.ts

```diff
@@ -0,0 +1,39 @@
+import { InventoryRecord, InventoryUpdateRecord, PurchaseRequestRecord, SiteRecord } from '../airtable/interface';
+import { getAllProducts, getInventorysByIds, getInventoryUpdatesByIds, getPurchaseRequestsByIds } from '../airtable/request';
+import { saveInventoryData, setLoadingForInventoryData } from './inventoryDataSlice';
+import { store } from './store';
+
+const refreshInventoryData = async (currentSite: SiteRecord): Promise<void> => {
+  store.dispatch(setLoadingForInventoryData());
+  if (currentSite) { // TODO @wangannie what to do if site null
+    const products = await getAllProducts();
+
+    const currentSiteInventory = await getInventorysByIds(currentSite.inventoryIds);
```

Comment:
Is there a reason you're making the extra request on the frontend instead of gathering the inventory data on the backend when pulling sites (aka pulling all the inventory data inside of the siteData access resolver)?

If there's good grounds for it, that's fine, but if not I'd prefer it done in the `Sites` access resolver on the backend because

1. It's faster (1 API call instead of 2)
2. I'm a little weary about trusting the internet connection for Users being flaky. I'd rather do the majority of API requests on the backend as a safety measure since the backend is under a more reliable environment.

If you have reasons for the frontend, that's fine just wanted to let you know the justification for the current layout